crates/wasmtime/src/component/func/typed.rs

```diff
@@ -730,6 +510,27 @@ forward_impls! {
     (T: Lower) Vec<T> => [T],
 }
 
+macro_rules! forward_lifts {
+    ($($a:ty,)*) => ($(
+        unsafe impl Lift for $a {
+            fn lift(store: &StoreOpaque, options: &Options, src: &Self::Lower) -> Result<Self> {
+                Ok(<WasmStr as Lift>::lift(store, options, src)?.to_str_from_store(store)?.into())
+            }
+
+            fn load(memory: &Memory<'_>, bytes: &[u8]) -> Result<Self> {
+                Ok(<WasmStr as Lift>::load(memory, bytes)?.to_str_from_memory(memory.as_slice())?.into())
+            }
+        }
+    )*)
+}
+
+forward_lifts! {
+    Box<str>,
+    std::rc::Rc<str>,
+    std::sync::Arc<str>,
+    String,
+}
```

Comment:
While you're here and doing this could this also add variants for lifting `Vec<T>` from `WasmSlice<T>` and such?

The macro could possibly look something like:

```rust
forward_lifts! {
    WasmStr => { Box<str>, Rc<str>, Arc<str>, String }
    WasmList<T> => { Box<[T]>, ...}
}
```

(or similar)